src/gg.regression.unity.bots/Runtime/Scripts/RGBotConfigs/RGStateProviders/RgStateHandlerPlatformer2DLevel.cs

```diff
@@ -0,0 +1,212 @@
+using UnityEngine;
+using UnityEngine.Tilemaps;
+
+namespace RegressionGames.RGBotConfigs.RGStateProviders
+{
+    [Serializable]
+    public class RgPlatformer2DPosition
+    {
+        public Vector2 position;
+        
+        // used by pathfinding to know if this has a wall adjoining it
+        public bool blockedRight;
+        // used by pathfinding to know if this has a wall adjoining it
+        public bool blockedLeft;
+
+        public override string ToString()
+        {
+            return $"Position: ({position.x}, {position.y})";
+        }
+    }
+
+    // ReSharper disable InconsistentNaming
+    [Serializable]
+    public class RgStateEntityBasePlatformer2DLevel : Dictionary<string,object>, IRGStateEntity
+    {
+        /**
+         * <summary>The BoundsInt of the whole tileMap</summary>
+         */
+        public BoundsInt tileMapBounds => (BoundsInt)
+            this.GetValueOrDefault("tileMapBounds", new BoundsInt(0,0,0,0,0,0));
+
+        /**
+         * <summary>The BoundsInt of the current visible portion of the tileMap</summary>
+         */
+        public BoundsInt currentBounds => (BoundsInt)
+            this.GetValueOrDefault("currentBounds", new BoundsInt(0,0,0,0,0,0));
+        
+        public Vector3 tileCellSize => (Vector3)this["tileCellSize"];
+        public RgPlatformer2DPosition[] platformPositions => (RgPlatformer2DPosition[])this["platformPositions"];
+        
+        public string GetEntityType()
+        {
+            return EntityTypeName;
+        }
+
+        public bool GetIsPlayer()
+        {
+            return false;
+        }
+        
+        public static readonly string EntityTypeName = "platformer2DLevel";
+        public static readonly Type BehaviourType = typeof(RgStateHandlerPlatformer2DLevel);
+        public static readonly bool IsPlayer = false;
+    }
+    
+    /**
+     * Provides state information about the tile grid in the current visible screen space.
+     */
+    [Serializable]
+    public class RgStateHandlerPlatformer2DLevel: RGStateBehaviour<RgStateEntityBasePlatformer2DLevel>
+    {
+        [Tooltip("How many tile spaces above a platform to consider when determining the height available on top of a platform.  This should match the height of the largest character model navigating the scene in tile units.")]
+        [Min(1)]
+        public int tileSpaceAbove = 2;
+        
+        [NonSerialized]
+        [Tooltip("Draw debug gizmos for platform locations in editor runtime ?")]
+        public bool renderDebugGizmos = true;
+
+        internal Vector3 _lastCellSize = Vector3.one;
+        private List<RgPlatformer2DPosition> _lastPositions = new();
+
+        protected override RgStateEntityBasePlatformer2DLevel CreateStateEntityInstance()
+        {
+            return new RgStateEntityBasePlatformer2DLevel();
+        }
+
+        public override void PopulateStateEntity(RgStateEntityBasePlatformer2DLevel stateEntity)
```

Comment:
Might use this as inspiration for my puzzle experiment since the state there is similar to the cells in a platformer

Reply:
remember to tag these (non-blocking) when applicable